### ISSUE TITLE
This website is under heavy load (queue full) - again

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/operations#557]
> This website is under heavy load (queue full)

> We're sorry, too many people are accessing this website at the same time. We're working on this problem. Please try again later.

For example on https://www.openstreetmap.org/changeset/108639533

What is the reason for that error appearing from time to time for all users? Servers not powerful enough? Scaling limitations? Some hard to track down bug?

**Comment by tomhughes** [comment]
The reason is unknown but under investigation - basically we are seeing load spikes but it's unclear if that is due to API usage spikes or some internal issues.

**Comment by mmd-osm** [comment]
Something odd here. Why would a frontend server like spike-08 show 65% system cpu time, which is roughly 20 cores only doing kernel work. It'd be very interesting to see what the kernel is doing here.

![spike-08](https://user-images.githubusercontent.com/5842757/127701338-163aac73-e270-4d80-9c8e-4d30d3225570.png)

**Comment by tomhughes** [comment]
Are you saying they're all doing that, or just spike-08?

**Comment by tomhughes** [comment]
It's all of them, but presumably that is just correlated with the request spikes we're seeing, so is it processing the requests.

I got as far as deciding the requests were from the "web" category but I need to do some work to try and narrow it down - my bet is on GPS traces though.

**Comment by mmd-osm** [comment]
Yes, it's all of them. spike-06 and spike-07 are showing the same pattern.

https://prometheus.openstreetmap.org/d/Ea3IUVtMz/host-overview?orgId=1&var-instance=spike-07&from=1627607570724&to=1627613589142&viewPanel=25

https://prometheus.openstreetmap.org/d/Ea3IUVtMz/host-overview?orgId=1&var-instance=spike-06&from=1627607570724&to=1627613589142&viewPanel=25

**Comment by mmd-osm** [comment]
> my bet is on GPS traces though.

This seems quite plausible to me.

**Comment by mmd-osm** [comment]
Seems to be less of an issue at the moment. https://uptime.openstreetmap.org/ suggests "Website HTTPS" is up for 70 days now.

**Comment by matkoniecz** [comment]
I have not seen it in some time.

**Comment by matkoniecz** [comment]
it is appearing again

**Comment by tomhughes** [comment]
I don't see any signs of significant increased load, and it seems fine now, so likely there was just a very brief spike that cause the queues to fill.

**Comment by tomhughes** [comment]
Actually spike-07 might be having issues - somebody is probably hitting it with something silly.
